src/formatDate.js

```diff
@@ -8,7 +8,40 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // Створюємо копії fromFormat і toFormat, щоб не змінювати параметри функції
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+
+  // Отримуємо роздільник з копії fromFormat (останній елемент масиву)
+  const fromSeparator = fromFormatCopy.pop();
+  const toSeparator = toFormatCopy.pop();
+
+  // Розбиваємо дату на частини за роздільником
+  const dateParts = date.split(fromSeparator);
+
+  // Створюємо об'єкт, який зіставляє формат з частинами дати
+  const dateObj = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  // Конвертуємо з YY в YYYY, якщо потрібно
+  if (dateObj.YY && !dateObj.YYYY) {
+    const year = parseInt(dateObj.YY, 10);
+
+    dateObj.YYYY = year < 30 ? '20' + dateObj.YY : '19' + dateObj.YY;
+  }
```

Comment:
The conversion from 'YY' to 'YYYY' is handled correctly. However, since you are using the `parseInt` function, you can directly assign the result to `dateObj.YYYY` without reusing `dateObj.YY` as a string. This will avoid the need for string concatenation and make the intent clearer.